demos/LowAllocationWebServer/ByteSpanExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+
+namespace LowAllocationServer
+{
+    public static class ByteSpanExtensions
+    {
+        internal static ByteSpan SliceTo(this ByteSpan buffer, char terminator, out int consumedBytes)
+        {
+            return buffer.SliceTo((byte) terminator, out consumedBytes);
```

Comment:
This cast might not be what we want. @krwq, do we have APIs to safely cast chars to Utf8CodeUnits?